### PR TITLE
CI: update docker-image ci to build multi-platform images (CON-1371)

### PR DESCRIPTION
**Pull request by Will-Panton**
## Description

This pull request enhances the Docker image CI workflow to include both `arm64` and `amd64` images for the ESP Matter development Docker image. Previously, as of January 10, 2023, the `arm64` build was discontinued due to space limitations on GitHub runners when building for multiple architectures sequentially on a single runner. This update resolves the space issue by adopting a [matrix approach](https://docs.docker.com/build/ci/github-actions/multi-platform/#distribute-build-across-multiple-runners) as recommended by Docker, which distributes the build process for each platform across multiple runners.

The motivation to also build an arm64 image comes from wanting to develop code inside a container on an Apple Silicon machine.

## Related

Screenshot 2024-10-10 at 11 54 21

See example workflow run [here](https://github.com/Will-Panton/esp-matter/actions/runs/11271170382).

## Testing

Tested the changes by forking and commenting out '# Disable the job in forks'. I ran the CI workflow with the docker credentials connected to my personal docker account. The resulting images can be seen [here](https://hub.docker.com/repository/docker/williampanton/esp-matter/tags). I then tested automated build on two runners, one ubuntu amd64, and a self hosted macOS arm64 runner using the new docker images which resulted in success.

---

## Checklist

Before submitting a Pull Request, please ensure the following:

- [x] 🚨 This PR does not introduce breaking changes.
- [x] All CI checks (GH Actions) pass.
- [x] Documentation is updated as needed.
- [x] Tests are updated or added as necessary.
- [x] Code is well-commented, especially in complex areas.
- [x] Git history is clean — commits are squashed to the minimum necessary.

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -38,12 +35,80 @@ jobs:
     if: ${{ github.repository_owner == 'espressif' }}
 
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
```
[the remark this answers is not in the document]
Comment (Will-Panton) [reply .github/workflows/docker-image.yml]: Agreed, previous images still available like you say.

**Review by shubhamdp — APPROVED** [review]

**Comment by shubhamdp** [comment]
@Will-Panton can you please squash the commits.

**Comment by Will-Panton** [comment]
> @Will-Panton can you please squash the commits.

Done ✅

**Comment by shubhamdp** [comment]
@Will-Panton PR is merged! Thank you for your contribution!

**Pull request merged**